pkg/log/zap.go

```diff
@@ -28,14 +31,89 @@ import (
 // logger of your application, and provide it to your components, by fx or manually.
 func NewDefaultZapLogger() (logr.Logger, error) {
 	// change the configuration in the future if needed.
-	zapLogger, err := zap.NewDevelopment()
+	envLevel := os.Getenv("LOG_LEVEL")
+	logLevel := zap.InfoLevel
+	if envLevel != "" {
+		var err error
+		logLevel, err = zapcore.ParseLevel(envLevel)
+		if err != nil {
+			return logr.Discard(), err
+		}
+	}
+
+	envMaxFieldSize := os.Getenv("LOG_MAX_FIELD_SIZE")
+	maxFieldSize := 8000
+	if envMaxFieldSize != "" {
+		var err error
+		maxFieldSize, err = strconv.Atoi(envMaxFieldSize)
+		if err != nil {
+			return logr.Discard(), err
+		}
+	}
+
+	encoderConfig := zapcore.EncoderConfig{
+		MessageKey:     "message",
+		LevelKey:       "level",
+		TimeKey:        "@timestamp",
+		NameKey:        "logger",
+		CallerKey:      "caller",
+		FunctionKey:    zapcore.OmitKey,
+		StacktraceKey:  "stacktrace",
+		LineEnding:     zapcore.DefaultLineEnding,
+		EncodeLevel:    zapcore.LowercaseLevelEncoder,
+		EncodeTime:     zapcore.RFC3339NanoTimeEncoder,
+		EncodeDuration: zapcore.SecondsDurationEncoder,
+		EncodeCaller:   zapcore.ShortCallerEncoder,
+		NewReflectedEncoder: func(w io.Writer) zapcore.ReflectedEncoder {
+			enc := json.NewEncoder(newTruncatingWriter(w, maxFieldSize))
```

Comment:
To reduce the change in behaviour, potentially the truncating writer could only be used if `LOG_MAX_FIELD_SIZE` is set - if the environment variable isn't found the default writer could be used.